#160451620 Role based Functionality

obulaworld opened the pull request:
#### What does this PR do?
- Implement admin role functionalities
   
#### Description of Task to be completed?
- write tests for endpoints
- create AdminController
  - add following methods  to AdminController   
      - suspendUser 
     - fetchAll
     - makeAdmin
     - stripAdmin

- create endpoints
    - list all admins
   `GET /api/v1/admin/all`
    - suspend a user
   `GET /api/v1/admin/suspend/:id`
   - change user role to admin
      ` GET /api/v1/admin/:id -  params(body) id, slug `
  - change admin role to user
   `GET /api/v1/admin/:id`

#### How should this be manually tested?

   request  GET /api/v1/admin/all
   ```js

response {
    "status": "success",
    "message": "Request Successfull",
    "users": [
        {
            "id": 1,
            "roleId": 1,
            "firstname": "Okafor",
            "lastname": "Emmanuel",
            "username": "DevOps",
            "email": "nondefyde@gmail.com",
            "password": "$2b$10$CjtrRPcpcHOYTRp4Zyd41O07uaodxjsx09ysb8pSgazovRh60ci1a",
            "socialId": null,
            "provider": null,
            "bio": null,
            "favouriteTags": null,
            "isPremium": null,
            "isSuspended": false,
            "imageUrl": null,
            "isActivated": false,
            "createdAt": "2018-10-18T14:03:33.295Z",
            "updatedAt": "2018-10-18T14:03:33.296Z"
        }
    ]
}
```
 GET /api/v1/admin/suspend/:id
   ```js
response {
    "status": "success",
    "message": "User Suspended Successfully",
    "user": [
        1
    ]
}
```

GET /api/v1/admin/role/change/admin/:id
request 

```js
response
{
   "status": "success",
    "message": "User role changed successfully",
    "user": [
        1
    ]
}
```

GET /api/v1/admin/role/change/user/:id

```js
response
{
   "status": "success",
    "message": "User role changed successfully",
    "user": [
        1
    ]
}
```
#### Any background context you want to provide?
 NA

#### What are the relevant pivotal tracker stories?
[#160451620](https://www.pivotaltracker.com/story/show/160451620 "role functionality")
 
### Question
 NA

### Screenshot
 NA

Commit:
chore(role functionality): Role based functiomnality
- Implements all admin functionality
closes #160451620

KaiserPhemi reviewed (changes requested):
#### Nitpick
Rebase and fix merge conflicts

obulaworld committed:
feat(role) Role Functionality

This feature enables different admin functionality such as
-Admin can suspend a user
- Admin can view all admins
-cAdmin can change user role

Closes #160451620

KaiserPhemi commented on swagger.yaml:
```diff
@@ -0,0 +1,158 @@
+## Reference => https://github.com/DinmaOtutu/RIDE-MY-WAY/
```
#### Critical
Can you please explain why we have this reference here?

obulaworld replied:
Good morning, Sorry about that but that file was not supposed to be pushed up. That reference has been removed in my current branch that implements the Swagger UI. It will come with my next PR.

obulaworld replied:
Good morning, Sorry about that but that file was not supposed to be pushed up. That reference has been removed in my current branch that implements the Swagger UI. It will come with my next PR.

KaiserPhemi commented on swagger.yaml:
```diff
@@ -0,0 +1,158 @@
+## Reference => https://github.com/DinmaOtutu/RIDE-MY-WAY/
```
#### Critical
Can you please explain why we have this reference here?

obulaworld replied:
File has been removed. It will come with the next PR

KaiserPhemi commented on swagger.yaml:
```diff
@@ -0,0 +1,158 @@
+## Reference => https://github.com/DinmaOtutu/RIDE-MY-WAY/
```
#### Critical
Can you please explain why we have this reference here?

obulaworld replied:
File has been removed. It will come with the next PR

KaiserPhemi commented on swagger.yaml:
```diff
@@ -0,0 +1,158 @@
+## Reference => https://github.com/DinmaOtutu/RIDE-MY-WAY/
```
#### Critical
Can you please explain why we have this reference here?

obulaworld replied:
File has been removed. It will come with the next PR

KaiserPhemi commented on swagger.yaml:
```diff
@@ -0,0 +1,158 @@
+## Reference => https://github.com/DinmaOtutu/RIDE-MY-WAY/
```
#### Critical
Can you please explain why we have this reference here?

KaiserPhemi replied:
#### Critical
Can you please explain why we have this reference here?

KaiserPhemi replied:
#### Critical
Can you please explain why we have this reference here?

KaiserPhemi reviewed (changes requested):
#### Critical
Please attend to the feedback.

obulaworld committed:
feat(role) Role Functionality)

This feature enables different admin functionality such as
-Admin can suspend a user
- Admin can view all admins
-cAdmin can change user role

Closes #160451620
